aoc/d3/d3_test.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import "testing"
+
+// ensure the strings given get split in two and returned as
+// the first and second items in a slice
+func TestSplitString(t *testing.T) {
+	tests := []struct {
+		input   string
+		output1 string
+		output2 string
+	}{
+		{
+			input:   "abcd",
+			output1: "ab",
+			output2: "cd",
+		},
+		{
+			input:   "vJrwpWtwJgWrhcsFMMfFFhFp",
+			output1: "vJrwpWtwJgWr",
+			output2: "hcsFMMfFFhFp",
+		},
+		{
+			input:   "jqHRNqRjqzjGDLGLrsFMfFZSrLrFZsSL",
+			output1: "jqHRNqRjqzjGDLGL",
+			output2: "rsFMfFZSrLrFZsSL",
+		},
+	}
+
+	for _, test := range tests {
+		result := SplitString(test.input)
+		if result[0] != test.output1 {
+			t.Errorf("Got %s, expected %s", result, test.output1)
+		}
+		if result[1] != test.output2 {
+			t.Errorf("Got %s, expected %s", result, test.output2)
+		}
+	}
+}
+
+// ensure the function returns the character that appears in both strings
+func TestFindCommonItem(t *testing.T) {
+	tests := []struct {
+		input1   string
+		input2   string
+		expected string
+	}{
+		{
+			input1:   "abc",
+			input2:   "cde",
+			expected: "c",
+		},
+		{
+			input1:   "vJrwpWtwJgWr",
+			input2:   "hcsFMMfFFhFp",
+			expected: "p",
+		},
+		{
+			input1:   "jqHRNqRjqzjGDLGL",
+			input2:   "rsFMfFZSrLrFZsSL",
+			expected: "L",
+		},
+	}
+
+	for _, test := range tests {
+		result := FindCommonItem(test.input1, test.input2)
+		if result != test.expected {
```

Comment:
might as well inline it

Reply:
Oooh didn't know you could declare a var inside an if. Fancy. Done in https://github.com/lucasrijllart/advent-of-code-2022/pull/1/commits/85b685bb038e687cf47423e10f0b78c4b5ba021a

Reply:
The power of the walrus. These days you can do this in Python too